src/Polyhedron.cu

```diff
@@ -7,11 +8,45 @@ __device__ Polyhedron::Polyhedron(Face *faces, int n_of_faces) :
 
 }
 
+__device__ Polyhedron &Polyhedron::operator=(const Polyhedron &other)
+{
+    if(this != &other)
+    {
+        faces = malloc_and_copy(other.faces, other.n_of_faces);
+        n_of_faces = other.n_of_faces;
+    }
+    return *this;
+}
+
+__device__ Polyhedron::Polyhedron(const Polyhedron &other)
+{
+    *this = other;
+}
+
+__device__ Polyhedron &Polyhedron::operator=(Polyhedron &&other) noexcept
+{
+    if(this != &other)
+    {
+        faces = nullptr;
```

Comment:
`Polyhedron` B has to be moved to `Polyhedron` A. If A was created earlier (move constructor is not called during the moving), then setting `A.faces` to `nullptr` will cause leak of memory that `A.faces` contained before moving B to A. Same to `Face::vertices`, `MapNode::particle` (I suppose) and `SimulationMap::nodes`

Reply:
Yes, you are right (by the way, I think, your explanation could be easier to understand if you'd provide a code snippet with an example). And yes, this does work for all `Polyhedron::faces`, `Face::vertices`, `MapNode::particle` and `SimulationMap::nodes`. Thank you for the alarm!